src/main/java/org/example/MiniProject2StudentManagementSystem/MainApplication.java

```diff
@@ -0,0 +1,135 @@
+package org.example.MiniProject2StudentManagementSystem;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Calendar;
+
+public class MainApplication {
+
+    public static void main(String[] args) {
+
+        Student student1 = new Student("S101", "John Doe");
+        Student student2 = new Student("S102", "Jane Doe");
+
+
+        Course course1 = new Course("C101", "Mathematics");
+        Course course2 = new Course("C102", "Programming");
+
+
+        enrollStudentInCourse(student1, course1);
+        enrollStudentInCourse(student1, course2);
+        enrollStudentInCourse(student2, course2);
+
+
+        displayStudentDetails(student1);
+        displayCourseDetails(course2);
+
+
+        List<Student> students = new ArrayList<>();
+        students.add(student2);
+        students.add(student1);
+
+        MainApplication.sortStudentsByName(students);
+        System.out.println("Students sorted by name:");
+        students.forEach(System.out::println);
+
+
+        String studentNameToSearch = "John Doe";
+        Student foundStudent = MainApplication.searchStudentByName(students, studentNameToSearch);
+        System.out.println("Found student by name '" + studentNameToSearch + "': " + foundStudent);
+
+
+        User adminUser = new Admin("admin", "admin123");
+        User teacherUser = new Teacher("teacher", "teacher123");
+        User studentUser = new StudentUser("student", "student123");
+
+
+        displayUserRole(adminUser);
+        displayUserRole(teacherUser);
+        displayUserRole(studentUser);
+    }
+
```

Comment:
none of the functions below shoud be static.

Reply:
I can see that this is work in progress but we should try separating these pieces of logic from the start at least a bit